src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/JCloudsProvider.java

```diff
@@ -129,9 +128,11 @@ public void deleteInstance(Infrastructure infrastructure, String instanceId) {
         try {
             execResponse = getComputeServiceFromInfastructure(infrastructure).runScriptOnNode(instanceId,
                                                                                               buildScriptToExecuteString(instanceScript),
-                                                                                              buildScriptOptions(instanceScript));
+                                                                                              buildScriptOptionsWithInstanceId(instanceScript,
+                                                                                                                               instanceId,
+                                                                                                                               infrastructure));
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            throw new RuntimeException("Script cannot be run on instance with id: " + instanceId, e);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00112)